### ISSUE TITLE
Is KL loss also useful for one-stage detectors like retinanet?

### ISSUE THREAD

**Issue by v-qjqs** [yihui-he/KL-Loss#31]
Hi, thanks for your nice open source project.
I'm doing experiments on retinanet with KL loss using my own customized dataset.
Since you have improved the performance of two-stage detectos like FasterRCNN and FPN under common datasets like COCO,
I wonder is KL loss also helpful for one-stage detectors like retinanet with COCO dataset?

And does the weights initialization of uncertainty prediction branch matters? Since FC is used for your two-stage detectors while conv2d is used for my retinanet experiments.

Thanks a lot.

**Comment by ethanhe42** [comment]
you can refer to yolo version https://github.com/yihui-he/KL-Loss#pytorch-re-implementations

yes, it helps training converge faster

**Comment by merlinarer** [comment]
> you can refer to yolo version https://github.com/yihui-he/KL-Loss#pytorch-re-implementations
> 
> yes, it helps training converge faster

Hello, the KL loss in Stronger yolo seems different from the paper. 
`elif cfg.boxloss == 'KL':
        l1_loss = respond_bbox * bbox_loss_scale * (
                torch.exp(-pred_vari) * smooth_loss(target=label_coor, input=pred_coor) + 0.5 * pred_vari) * cfg.l1scale
        bbox_loss = l1_loss `
This code only includes loss_bbox and bbox_pred_std_abs_logw_loss ,bbox_pred_std_abs_mulw_loss is not there.
